### ISSUE TITLE
[BUG] Additional double quotes added to serialization of bearer token by tapir

### ISSUE THREAD

**Issue by pbuszka** [softwaremill/tapir#1323]
Tapir version: 0.17.19

Scala version: 2.13.5

For endpoint with bearer token extraction (the exact endpoint definition does not matter)
```
// endpoint definition
  val baseEndpoint = endpoint
    .in("api")

  val authenticateHouse: Endpoint[String, String, House, Any] = baseEndpoint
    .get
    .in("authorize")
    .in(TapirAuth.bearer[String]())
    .out(jsonBody[House])
    .errorOut(stringBody)

// server side interpreter
AkkaHttpServerInterpreter.toRoute(BackendServerEndpoint.authenticateHouse)

// client side request
val backend = FetchMonixBackend()
val request = SttpClientInterpreter.toRequest(BackendEndpoint.authenticateHouse, Some(uri"http://localhost:8080"))
backend.send(request("token"))

```
The request generated via client interpreter works flawlessly but I tried to use curl to access the endpoint and it fails when decoding Authorization header.
```
curl -i localhost:8080/api/authorize -H 'Authorization: Bearer token'
HTTP/1.1 400 Bad Request
Server: akka-http/10.2.4
Date: Wed, 16 Jun 2021 15:11:30 GMT
Content-Type: text/plain; charset=UTF-8
Content-Length: 39

Invalid value for: header Authorization
```
However this request passes (note the double quotes around token)
```
curl -i localhost:8080/api/authorize -H 'Authorization: Bearer "token"'
HTTP/1.1 200 OK
Server: akka-http/10.2.4
Date: Wed, 16 Jun 2021 15:12:01 GMT
Content-Type: application/json
Content-Length: 464
...
```

**How to reproduce?**

Define an endpoint and access it from curl as above

**Additional information**
I think it is linked somehow with serialization/deserialization which works well if both ends use tapir. When I sniffed in chrome the headers actually passed by tapir client it is `authorization: Bearer "token"` (note again the added quotes).

**Comment by slabiakt** [comment]
I couldn't reproduce this issue.  With this snippet:

```
object Main extends App {
  val helloWorld: Endpoint[String, Unit, String, Any] = endpoint.get.in("hello").in(TapirAuth.bearer[String]()).out(stringBody)
  val helloWorldRoute: Route = AkkaHttpServerInterpreter.toRoute(helloWorld)(token => Future.successful(Right(s"token is:  ${token}")))
  implicit val actorSystem: ActorSystem = ActorSystem()
  val bindAndCheck = Http().newServerAt("localhost", 8080).bindFlow(helloWorldRoute)
}
```

I get this output:

```
slabiakt@Tomaszs-MBP ~ % curl -i localhost:8080/hello -H 'Authorization: Bearer slabiak123'

HTTP/1.1 200 OK
Server: akka-http/10.2.4
Date: Thu, 17 Jun 2021 10:04:18 GMT
Content-Type: text/plain; charset=UTF-8
Content-Length: 21

token is:  slabiak123%                                                                                                                          
slabiakt@Tomaszs-MBP ~ % curl -i localhost:8080/hello -H 'Authorization: Bearer "slabiak123"'

HTTP/1.1 200 OK
Server: akka-http/10.2.4
Date: Thu, 17 Jun 2021 10:04:27 GMT
Content-Type: text/plain; charset=UTF-8
Content-Length: 23

token is:  "slabiak123"%                                                                                                                        
```

It was tested on:
MacOs Big Sur 
Tapir version: 0.17.19
Scala version: 2.13.6

Could you please test my snippet and let me know on what OS do you run it?

**Comment by pbuszka** [comment]
Your snippet works correctly in my config. I just realized that the real diference is not the endpoint but the handling of the codecs.  I generate Codecs for case classes based on upickle (actually my custom pickler based on upickle) with the following implict which was in scope for my endpoint

```
 implicit def readWriterTextPlainCodec[T: BSONPickler.ReadWriter: Schema]: PlainCodec[T] =
    Codec.string.map(BSONPickler.read(_))(BSONPickler.write(_)).schema(implicitly[Schema[T]])
```
and I have accidently shot myself in the foot with it. It seems that it overrides `Codec[List[String], * CodecFormat.TextPlain]` which is used by auth.bearer and hence the strange behaviour in my setup. So the bug was generated by the code on my side.

I only wonder if tapir should allow for such override in this particular case of auth.bearer (and other Authorization related stuff)  where the Codec should always be standard. Maybe a warning in the docs ?

In my case it was an unintended consequence but if somebody wanted to have custom `Codec[List[String], * CodecFormat.TextPlain]` and use auth.bearer at the same time then it is not possible without juggling the fragile implicits scopes. 

Anyway sorry for the false alarm

**Comment by adamw** [comment]
Ha that's surprising - are you not using `CodecFormat.Json` intentionally? The purpose of this third parameter is to differentiate between codecs in scenarios like this. 

Implicit precedence is something that's built into scala, so we have little control as when it comes to issuing warnings :)

**Comment by pbuszka** [comment]
As far as I remember I tried to cover some corner case with some broken external API and I didn't understand how it works at the time and probably messed up with those two CodecFormat versions. Then the main serialization moved to JSON but the old implicit stayed in the code.The problem didn't show up in my own code because I share tapir endpoint definition both client and server side and everything worked perfectly Ok. It came up when I tried to diagnose some authorization problem using curl.
